impeller/golden_tests/vulkan_screenshotter.mm

```diff
@@ -0,0 +1,90 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/impeller/golden_tests/vulkan_screenshotter.h"
+
+#include "flutter/fml/synchronization/waitable_event.h"
+#include "flutter/impeller/golden_tests/metal_screenshot.h"
+#include "impeller/renderer/backend/vulkan/surface_context_vk.h"
+#include "impeller/renderer/backend/vulkan/texture_vk.h"
+#define GLFW_INCLUDE_NONE
+#include "third_party/glfw/include/GLFW/glfw3.h"
+
+namespace impeller {
+namespace testing {
+
+namespace {
+std::unique_ptr<Screenshot> ReadTexture(
+    const std::shared_ptr<Context>& surface_context,
+    const std::shared_ptr<Texture>& texture) {
+  DeviceBufferDescriptor buffer_desc;
+  buffer_desc.storage_mode = StorageMode::kHostVisible;
+  buffer_desc.size =
+      texture->GetTextureDescriptor().GetByteSizeOfBaseMipLevel();
+  std::shared_ptr<DeviceBuffer> device_buffer =
+      surface_context->GetResourceAllocator()->CreateBuffer(buffer_desc);
+  FML_CHECK(device_buffer);
+
+  auto command_buffer = surface_context->CreateCommandBuffer();
+  auto blit_pass = command_buffer->CreateBlitPass();
+  bool success = blit_pass->AddCopy(texture, device_buffer);
+  FML_CHECK(success);
+
+  success = blit_pass->EncodeCommands(surface_context->GetResourceAllocator());
+  FML_CHECK(success);
+
+  fml::AutoResetWaitableEvent latch;
+  success =
+      command_buffer->SubmitCommands([&latch](CommandBuffer::Status status) {
+        FML_CHECK(status == CommandBuffer::Status::kCompleted);
+        latch.Signal();
+      });
+  FML_CHECK(success);
+  latch.Wait();
+
+  // TODO(gaaclarke): Replace CoreImage requirement with something
```

Comment:
Just checking I understand - this is OSX-only code for MoltenVK?

Reply:
In this case we are using swiftshader on macos.  The tests should fail if you use moltenvk, there's a check in the TearDown somewhere that makes sure libMoltenVK.dylib isn't loaded.